apps/calendar/js/views/modify_event.js

```diff
@@ -357,114 +256,69 @@ ModifyEvent.prototype = {
+    var moveDate = this.event.startDate;
 
-    function fetchProvider(err, result) {
-      provider = result;
-      provider.eventCapabilities(
-        self.event.data,
-        verifyCaps
-      );
-    }
+    try {
+      yield core.bridge[method](this.event);
+      // move the position in the calendar to the added/edited day
+      core.timeController.move(moveDate);
+      // order is important the above method triggers the building
+      // of the dom elements so selectedDay must come after.
+      core.timeController.selectedDay = moveDate;
 
-    function verifyCaps(err, caps) {
-      if (err) {
-        return console.error('Error fetching capabilities for', self.event);
-      }
+      // we pass the date so we are able to scroll to the event on the
+      // day/week views
+      var state = {
+        eventStartHour: moveDate.getHours()
+      };
 
-      // safe-guard but should not ever happen.
-      if (caps[capability]) {
-        persistEvent();
+      if (method === 'updateEvent') {
+        // If we edit a view our history stack looks like:
+        //   /week -> /event/view -> /event/save -> /event/view
+        // We need to return all the way to the top of the stack
+        // We can remove this once we have a history stack
+        viewFactory.get('ViewEvent', v => router.go(v.returnTop(), state));
+        return;
       }
-    }
 
-    function persistEvent() {
-      var list = self.element.classList;
+      router.go(this.returnTo(), state);
 
-      // mark view as 'in progress' so we can style
-      // it via css during that time period
-      list.add(self.PROGRESS);
-
-      var moveDate = self.event.startDate;
-
-      provider[method](self.event.data, function(err) {
-        list.remove(self.PROGRESS);
-
-        if (err) {
-          self.showErrors(err);
-          return;
-        }
-
-        // move the position in the calendar to the added/edited day
-        core.timeController.move(moveDate);
-        // order is important the above method triggers the building
-        // of the dom elements so selectedDay must come after.
-        core.timeController.selectedDay = moveDate;
-
-        // we pass the date so we are able to scroll to the event on the
-        // day/week views
-        var state = {
-          eventStartHour: moveDate.getHours()
-        };
-
-        if (method === 'updateEvent') {
-          // If we edit a view our history stack looks like:
-          //   /week -> /event/view -> /event/save -> /event/view
-          // We need to return all the way to the top of the stack
-          // We can remove this once we have a history stack
-          viewFactory.get('ViewEvent', function(view) {
-            router.go(view.returnTop(), state);
-          });
-
-          return;
-        }
-
-        router.go(self.returnTo(), state);
-      });
+    } catch(err) {
+      this.showErrors(err);
     }
-  },
+
+    this.element.classList.remove(this.PROGRESS);
```

Comment:
you're right. I'll extract the `router.go()` and `returnTop()` into another method to solve this.